torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -117,12 +117,22 @@ def selective_checkpointing_context_fn():
 
 def get_tp_parallel_strategy(
     job_config: JobConfig,
+    model: nn.Module,
 ) -> Tuple[RowwiseParallel, ColwiseParallel, PrepareModuleInput]:
     """Get the parallel strategy for the transformer model.
 
     This function handles the special case of using float8 with tensor parallelism.
     """
-    if job_config.training.fp8_linear == "dynamic":
```

Comment:
`fp8_linear == "dynamic"` is outdated after recent unification of dynamic/delayed scaling (`Float8Linear`)
https://github.com/pytorch/torchtitan/pull/436 

update it in this PR to make TP fp8 all-gather work again